pkg/config/flock_unix.go

```diff
@@ -0,0 +1,60 @@
+//go:build unix
+
+/*
+Copyright 2025 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"golang.org/x/sys/unix"
+	"os"
+)
+
+// tryLock attempts to acquire the lock. Returns *os.File if successful, nil otherwise.
+func tryLock(path string) (*os.File, error) {
+	file, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR, 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := unix.Flock(int(file.Fd()), unix.LOCK_EX|unix.LOCK_NB); err != nil {
+		_ = file.Close()
+		if err == unix.EWOULDBLOCK {
+			return nil, ErrK0sAlreadyRunning // Lock is already held by another process
+		}
+		return nil, err
+	}
+	return file, nil
+}
+
+// locked checks if the lock is currently held by another process.
+func locked(path string) bool {
```

Comment:
This could be implemented in terms of `tryLock`, for both UNIX and Windows:

```go
func locked(path string) (bool, error) {
	f, err := tryLock(path)
	if err != nil {
		if errors.Is(err, ErrK0sAlreadyRunning) {
			return true, nil
		}
		return false, err
	}
	return false, f.Close()
}
```

Reply:
problem is that it will acquire an exclusive lock even for a very short time, if that happens while k0s is starting up, k0s could in theory fail to start up with a false "already running" lock. That's why I use a shared lock instead of exclusive to test if it is locked. I suppose we could pass an argument to tryLock() as well, but I tried to keep things simple.

Reply:
I generally don't like [boolean arguments](https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3), but ok if you prefer a `exclusive boolean` arg, then I'll add it so we can move this forward.